cmake/Modules/FindMKL.cmake

```diff
@@ -286,7 +279,8 @@ SET(CMAKE_LIBRARY_PATH ${saved_CMAKE_LIBRARY_PATH})
 SET(CMAKE_INCLUDE_PATH ${saved_CMAKE_INCLUDE_PATH})
 IF (MKL_LIBRARIES AND MKL_INCLUDE_DIR)
   SET(MKL_FOUND TRUE)
-  set(MKL_cmake_included true)
+  SET(CAFFE2_USE_MKL ON)
```

Comment:
Let's rely on MKL_FOUND instead of `CAFFE2_USE_MKL`

Reply:
Also, does all of this need to get upstreamed to Intel's repo?

Reply:
pls see blow comments.